### PR TITLE
[4.0] installer plugin detection

### PR DESCRIPTION
**Pull request by burakozcelik5028**
Installer Plugin Detection (Joomla4) #20608

#20608 

### Summary of Changes 
_FIRST STEP_
  (Joomla 3) We have four Plugins Installers:
1)Install From WEB 
2)Install From Upload 
3)Install From Folder 
4)Install From URL
 But System did not declera to 'Install From WEB'. When we had just sellected the 'Install From WEB', system give [warning message](http://www.imgim.com/1plugininstallererror-kopya.jpeg):
image

_SECOND STEP_
     (Joomla 4) When we disable these four plugins, System gave us [error table](http://www.imgim.com/1plugininstallererror.jpeg):
image

### Testing Instructions
_FIRST STEP_
We disabled 'Install from Upload', 'Install from Folder', 'Install from URL'. 
Extensions---->>Plugins--->Select Type--->Installer

When we control the way, we had warning. 
Extensions---->>Manage--->>Install

_SECOND STEP_
We disabled all installer plugins.
System--->>Manage-->>Plugins-->>Table Options--->>Select Type--->>Installer

We had warning table because of disabled installer plugins.
System--->>Install--->>Extensions

### Actual result
_FIRST STEP_
 Before Solve:
image

After Solve:
image

_SECOND STEP_
[(Before Solve)](http://www.imgim.com/1plugininstallererror.jpeg):
image

[(After Solve)](http://www.imgim.com/4aftersolvestep2.jpeg):
image

@brianteeman

### CODE REVIEW

**Comment by brianteeman** [comment]
~With no plugins enabled i get~

~Notice: Undefined offset: 0 in C:\htdocs\cms4\administrator\components\com_installer\tmpl\install\default.php on line 41~

**Comment by brianteeman** [comment]
a bit confused by your comment about having to install the install from web plugin
it is installed by default in joomla 4

**Comment by burakozcelik5028** [comment]
@brianteeman line 41 is `$active = '';` so it can not generate this notice.
Can you please check if the PR is applied correctly?

File: administrator/components/com_installer/tmpl/install/default.php [hunk 1]
```diff
@@ -36,9 +36,16 @@
 						<?php echo $this->loadTemplate('message'); ?>
 					<?php endif; ?>
 					<?php // Show installation tabs at the start ?>
-					<?php $firstTab = $app->triggerEvent('onInstallerViewBeforeFirstTab', array()); ?>
 					<?php $tabs = $app->triggerEvent('onInstallerAddInstallationTab', array()); ?>
-					<?php echo JHtml::_('bootstrap.startTabSet', 'myTab', array('active' => $tabs[0]['name'])); ?>
+					<?php
```
Comment (Quy) [inline administrator/components/com_installer/tmpl/install/default.php]: Code can be simplified to be:

```
<?php $active = isset($tabs[0]) ? $tabs[0]['name'] : ''; ?>
```

File: administrator/components/com_installer/tmpl/install/default.php [hunk 2]
```diff
@@ -37,14 +37,7 @@
...
 					<?php $tabs = $app->triggerEvent('onInstallerAddInstallationTab', array()); ?>
-					<?php
-					$active = '';
-
-					if (isset($tabs[0]))
-					{
-						$active = $tabs[0]['name'];
-					}
-					?>
+                    <?php $active = isset($tabs[0]) ? $tabs[0]['name'] : ''; ?>
```
Comment (Quy) [inline administrator/components/com_installer/tmpl/install/default.php]: Please use tabs and not spaces to indent.

**Comment by brianteeman** [comment]
@barbarosuntorunu sorry that was a silly unrelated issue from me

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 5446df0db8004d171e0afa70481233b43159861a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21262">issues.joomla.org/tracker/joomla-cms/21262</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 5446df0db8004d171e0afa70481233b43159861a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21262">issues.joomla.org/tracker/joomla-cms/21262</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

**Comment by laoneo** [comment]
Can you fix conflicts please.

**Comment by burakozcelik5028** [comment]
@laoneo Okay, I will do.

**Comment by laoneo** [comment]
Conflicts are still there ;-)

**Comment by roland-d** [comment]
@barbarosuntorunu Thank you for the pull, request, can you fix the merge conflicts?

**Comment by Quy** [comment]
Confirmed. OK in Firefox, but not in Chrome.

Click the checkbox next to the article.
Click on the article name to edit.
Click the Back button.

**Comment by Quy** [comment]
Replaced by PR #25238.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21262">issues.joomla.org/tracker/joomla-cms/21262</a>.</sub>

**Pull request closed without merging**